spring-cloud-stream-reactive/src/main/java/org/springframework/cloud/stream/reactive/StreamEmitterAnnotationBeanPostProcessor.java

```diff
@@ -0,0 +1,244 @@
+			if (targetMethod != null) {
+				Assert.isTrue(targetMethod.getAnnotation(Input.class) == null,
+						StreamEmitterErrorMessages.INPUT_ANNOTATIONS_ARE_NOT_ALLOWED);
+				String methodAnnotatedOutboundName = StreamAnnotationCommonMethodUtils.getOutboundBindingTargetName(targetMethod);
+				int outputAnnotationCount = StreamAnnotationCommonMethodUtils.outputAnnotationCount(targetMethod);
+				validateStreamEmitterMethod(targetMethod, outputAnnotationCount, methodAnnotatedOutboundName);
+				invokeSetupMethodOnToTargetChannel(targetMethod, targetBean, methodAnnotatedOutboundName);
+			}
+			this.running = true;
+		}
+	}
+
+	@SuppressWarnings({"rawtypes", "unchecked"})
+	private void invokeSetupMethodOnToTargetChannel(Method method, Object bean, String outboundName) {
+		Object[] arguments = new Object[method.getParameterTypes().length];
+		for (int parameterIndex = 0; parameterIndex < arguments.length; parameterIndex++) {
+			MethodParameter methodParameter = MethodParameter.forMethodOrConstructor(method, parameterIndex);
+			Class<?> parameterType = methodParameter.getParameterType();
+			Object targetReferenceValue = null;
+			if (methodParameter.hasParameterAnnotation(Output.class)) {
+				targetReferenceValue = AnnotationUtils.getValue(methodParameter.getParameterAnnotation(Output.class));
+			}
+			else if (arguments.length == 1 && StringUtils.hasText(outboundName)) {
+				targetReferenceValue = outboundName;
+			}
+			if (targetReferenceValue != null) {
+				Assert.isInstanceOf(String.class, targetReferenceValue, "Annotation value must be a String");
+				Object targetBean = this.applicationContext.getBean((String) targetReferenceValue);
+				for (StreamListenerParameterAdapter<?, Object> streamListenerParameterAdapter : this.streamListenerParameterAdapters) {
+					if (streamListenerParameterAdapter.supports(targetBean.getClass(), methodParameter)) {
+						arguments[parameterIndex] = streamListenerParameterAdapter.adapt(targetBean,
+								methodParameter);
+						break;
+					}
+				}
+				Assert.notNull(arguments[parameterIndex], "Cannot convert argument " + parameterIndex + " of " + method
+						+ "from " + targetBean.getClass() + " to " + parameterType);
+			}
+			else {
+				throw new IllegalStateException(StreamEmitterErrorMessages.ATLEAST_ONE_OUTPUT);
+			}
+		}
+		Object result;
+		try {
+			result = method.invoke(bean, arguments);
+		}
+		catch (Exception e) {
+			throw new BeanInitializationException("Cannot setup StreamEmitter for " + method, e);
+		}
+
+		if (!Void.TYPE.equals(method.getReturnType())) {
+			if (!StringUtils.hasText(outboundName)) {
+				for (int parameterIndex = 0; parameterIndex < method.getParameterTypes().length; parameterIndex++) {
+					MethodParameter methodParameter = MethodParameter.forMethodOrConstructor(method,
+							parameterIndex);
+					if (methodParameter.hasParameterAnnotation(Output.class)) {
+						outboundName = methodParameter.getParameterAnnotation(Output.class).value();
```

Comment:
Nit-pick, but a bit above in this method you use:
```
 AnnotationUtils.getValue(methodParameter.getParameterAnnotation(Output.class))
```
This one more addition to that my "obvious string" concern.